src/components/editor/markdown-renderer/markdown-renderer.tsx

```diff
@@ -127,6 +131,13 @@ export const MarkdownRenderer: React.FC<MarkdownRendererProps> = ({ content, onM
       })
     }
     md.use(taskList)
+    if (plantumlServer) {
```

Comment:
Hm, I'm a bit unsure about this if-statement.

Of course if no PlantUML server is defined, we can't render anything. But maybe we should show an error message in that case.

While for the admin of an instance it's clear that PlantUML is disabled for example, a user that tries to copy a document with a PlantUML diagram from another instance might be confused why it won't work on this instance.

Reply:
I think that PlantUML should be an optional, opt-in feature. It requires the admin to either setup their own PlantUML server or trust parts of their users documents to a third party server (http://www.plantuml.com/plantuml/).

Reply:
I agree that PlantUML should be opt-in. But I guess it's better anyway to show a message like "PlantUML is not enabled on this server" instead of a silent failure.

Reply:
The general syntax for plantUML is written between \```plantuml and \```. If the Plugin is not supported the normal rendering for code should be used so I think this is an appropriate  work around.
But it's also true that it's kinda puzzling if plantuml code get's rendered on one server and not on the next. Not sure how that could be reconciled with everything we have right now. An permanent error message above every plantuml code block, letting the user know that their admin did not activate this feature seems rather odd, too.
Maybe the best idea would be (if we extract the plantuml code into a hedgedoc plugin) to show the error when the plugin is included and the plantuml server is not configured, but not if the plugin is absent. I'd suggest opening a separate issue for this and mark this thread as resolved in the mean time.

Reply:
I don't think that we need to extract the code plantuml code. The markdown parser is initialized on every rendering (something we should probably not do). So if plantuml is activated and possible use the markdown it plugin. Otherwise include a markdown it plugin that replaces the fence with an alert.

Reply:
Then open an issue and link it here.

Reply:
Or let's ask @truh. Do you want to add the "plantuml is not activated" banner plugin here or in an extra pr?

Reply:
If I understand @DerMolly correctly, he doesn't want a banner plugin, he wants this PlantUML code be refactored into a plugin (as soon as a plugin interface is available) and the combination of _plugin is installed_ + _server is not configured_ should show an error message.

Reply:
If i understand him correcly, he wants to show a message that plantuml code was recognized, but can't be processed because the admin didn't activate plantuml. 
You can implement that easily. You can check if the planuml server url is set and if it is, then you use the plantuml markdown it plugin. If it's not activated, then you use the "show a plantuml is not activate" markdown it plugin.

Reply:
That's indepentend from a hedgedoc-frontend-plugin-api

Reply:
Showing a warning in the frontend just sounds like an anti-feature to me. If we are planning for the hedgedoc frontend to have plugin support there will be a lot of cases where rendering differs between instances and having code blocks as a fallback for unsupported languages seems preferable over warning boxes.

Reply:
I'm not talking about a general fallback for unsupported languages. I'm talking about a warning in this specific case. Something like "Plantuml rendering is installed, but the admin didn't set a plantuml server url. Therefore I can't render your diagram. sorry"

Reply:
Because then you want plantuml to work, but it doesn't and I guess it's good to let users know WHY it's not working.

Reply:
I do thing that stuff like this should be handled in a consistent manner.

The reason why something doesn't work should concern the admin of an installation, not the user. If we consider an empty plantumlServer option a configuration error, this should be logged to the server console, not show up on the frontend.

Reply:
If the user expects this to work and it doesn't work there are two reasons:
The user made a mistake or the admin did it. Why should we confuse the user if we can tell him that it's not his fault? I don't get what's bad with a notice that says "the admin has misconfigured plantuml"

Reply:
IMHO showing a warning should be included in this PR for now. When we extract this code into a plugin we'll need that anyway and it doesn't hurt to implement it now.